src/material/dialog/dialog-config.ts

```diff
@@ -43,6 +43,13 @@ export class MatDialogConfig<D = any> {
    */
   viewContainerRef?: ViewContainerRef;
 
+  /**
+   * Injector used for the instantiation of the component to be attached.
+   * Note: the `viewContainerRef` injector has a higher priority than the `injector`.
+   * The dialog will first read the `viewContainerRef` injector and then fall back to this injector.
+   */
+  injector?: Injector;
```

Comment:
I think the use-case is reasonable, but I'd definitely prefer to hear from @mmalerba / @crisbeto 

Also, I find it a little confusing that the view container ref would take precedence. If I explicitly provide an injector, I would expect that one to be used instead of the `ViewContainerRef`-injector.